### ISSUE TITLE
r2 not ok for log-log

### ISSUE THREAD

**Issue by jonreiter** [100trillionUSD/bitcoin#3]
it's not really ok to run the (linear) r2 for a log-log plot.  i've added a page which produces a proper, similar, measure in my fork:
https://github.com/jonreiter/bitcoin/blob/master/sf_proper_r2.xls
that comes in at 0.63.

r^2 on the linear log-log plot can make even bad models look good.  google will return you many articles on this.  your charts are surely interesting and there is something here but those stats are not quite right.

**Comment by phraudsta** [comment]
you are mistaken. in your sheet you fit logest on market price and sf. logest estimates the model form y = b*m^x (i.e. an exponential form).

**Comment by jonreiter** [comment]
exponential on untransformed values yes.  that's the same form as linest on log-log values.  but the error should be measured untransformed.

the initial post even references that exact form under the name "power law." that form is correct for sure.

**Comment by phraudsta** [comment]
no, logest is not estimating the power law that way. it is estimating exponential growth. for instance, I direct you to [https://docs.microsoft.com/en-us/office/troubleshoot/excel/statistical-functions-logest](https://docs.microsoft.com/en-us/office/troubleshoot/excel/statistical-functions-logest) which gives us the algorithm logest uses:

> You call LOGEST(known_y's, known_x's, constant, TRUE).
> LOGEST calls LINEST(LN of known_y's, known_x's, constant, TRUE).
> LOGEST receives the results table from this call to LINEST.
> LOGEST modifies LINEST coefficients in the first row of LINEST result table by exponentiation. 

Thus to correctly estimate the power law using logest, you would call logest(value, ln(sf), true, true)
or in your spreadsheet, alter to: =LOGEST(C11:C124,D11:D124,1,1)

the r2 remains at 0.95.

**Comment by 100trillionUSD** [comment]
yes. it is just a linear regression, log(price)= a+b*log(S2F-ratio), so I think R2 applies just fine. but I rewrote the formula to non-log: price=a*S2F^b so that might be confusing.
